### PR TITLE
Add Person ID partition key generator

### PR DESCRIPTION
**Pull request by pbheemag**
## Description

Adds a new person ID partition key generator. This makes sense for users of person ID to have an even more granular way to store attributes. If person ID is unavailable, the key will fall back to using the user ID.

## Testing

New unit tests.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Docs(Add new document content)
- [ ] Translate Docs(Translate document content)

## Checklist

- [x] My code follows the code style of this project
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [x] I have read the **README** document
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed
- [x] My commit message follows [Conventional Commit Guideline](https://conventionalcommits.org/)

## License

- [x] I confirm that this pull request can be released under the Apache 2 license

[issues]: https://github.com/alexa/alexa-skills-kit-sdk-for-nodejs/issues
[license]: http://aws.amazon.com/apache2.0/
[cla]: http://en.wikipedia.org/wiki/Contributor_License_Agreement

### CODE REVIEW

File: ask-sdk-dynamodb-persistence-adapter/lib/attributes/persistence/PartitionKeyGenerators.ts [hunk 1]
```diff
@@ -62,4 +62,21 @@ export const PartitionKeyGenerators = {
 
         return requestEnvelope.context.System.device.deviceId;
     },
+
+    /**
+     * Gets attributes id using person id.
```
Comment (nikhilym) [inline ask-sdk-dynamodb-persistence-adapter/lib/attributes/persistence/PartitionKeyGenerators.ts:67]: This misses out the fallback information of returning user id if person id is not present. Can you add that as well?

**Review by nikhilym — CHANGES_REQUESTED** [review]

File: ask-sdk-dynamodb-persistence-adapter/lib/attributes/persistence/PartitionKeyGenerators.ts [hunk 2]
```diff
@@ -62,4 +62,21 @@ export const PartitionKeyGenerators = {
...
+     * @param {RequestEnvelope} requestEnvelope
+     * @returns {string}
+     */
+    personId(requestEnvelope : RequestEnvelope) : string {
+        if (requestEnvelope
+              && requestEnvelope.context
+              && requestEnvelope.context.System
+              && requestEnvelope.context.System.person
+              && requestEnvelope.context.System.person.personId) {
```
Comment (ShenChen93) [inline ask-sdk-dynamodb-persistence-adapter/lib/attributes/persistence/PartitionKeyGenerators.ts]: Please remove the trailing whitespace

File: ask-sdk-dynamodb-persistence-adapter/lib/attributes/persistence/PartitionKeyGenerators.ts [hunk 3]
```diff
@@ -62,4 +62,21 @@ export const PartitionKeyGenerators = {
...
+    personId(requestEnvelope : RequestEnvelope) : string {
+        if (requestEnvelope
+              && requestEnvelope.context
+              && requestEnvelope.context.System
+              && requestEnvelope.context.System.person
+              && requestEnvelope.context.System.person.personId) { 
+            return requestEnvelope.context.System.person.personId;
+        }
+
+        return this.userId(requestEnvelope);
```
Comment (ShenChen93) [inline ask-sdk-dynamodb-persistence-adapter/lib/attributes/persistence/PartitionKeyGenerators.ts]: According to our tslint rules, the "this" keyword is disallowed outside of a class body

File: ask-sdk-dynamodb-persistence-adapter/lib/attributes/persistence/PartitionKeyGenerators.ts [hunk 4]
```diff
@@ -62,4 +62,21 @@ export const PartitionKeyGenerators = {
...
+    }
```
Comment (ShenChen93) [inline ask-sdk-dynamodb-persistence-adapter/lib/attributes/persistence/PartitionKeyGenerators.ts]: missing trailing comma

**Review by ShenChen93 — COMMENTED** [review]
Hi @pbheemag ,

Thanks for the PR ! The PR looks good to me except some lint issues. Also, since person attribute is added to RequestEnvelope.context.system after ask-sdk-model v1.23.0, it would be great if you could update the the dependency version of ask-sdk-model to latest v1.24.0 in package.json as well. Otherwise it may fail the travis check.

Thanks,
Shen

**Review by ShenChen93 — COMMENTED** [review]
Hi @pbheemag ,

Thanks for the PR ! The PR looks good to me except some lint issues. Also, since person attribute is added to RequestEnvelope.context.system after ask-sdk-model v1.23.0, it would be great if you could update the the dependency version of ask-sdk-model to latest v1.24.0 in package.json as well. Otherwise it may fail the travis check.

Thanks,
Shen

File: ask-sdk-dynamodb-persistence-adapter/lib/attributes/persistence/PartitionKeyGenerators.ts (hunk 3 above)
Comment (pbheemag) [reply ask-sdk-dynamodb-persistence-adapter/lib/attributes/persistence/PartitionKeyGenerators.ts]: A little concerned about backward compatibility if we were to change this file to a class format. Is there any other way of achieving this?

**Review by ShenChen93 — APPROVED** [review]

**Review by nikhilym — APPROVED** [review]

**Pull request merged**